Content.Shared/Chemistry/EntitySystems/SharedSolutionContainerSystem.cs

```diff
@@ -452,6 +454,12 @@ public bool RemoveReagent(Entity<SolutionComponent> soln, ReagentQuantity reagen
         var (uid, comp) = soln;
         var solution = comp.Solution;
 
+        // prevent issues where removing reagents from shared causes the amount to be removed multiple times.
+        if (!GameTiming.IsFirstTimePredicted)
+        {
+            return solution.GetReagentQuantity(reagentQuantity.Reagent) >= reagentQuantity.Quantity;
+        }
+
```

Comment:
That's the point of prediction? This looks like a bandaid unless the state handler is also calling this.